curio/thread.py

```diff
@@ -67,10 +67,10 @@
 
 class AsyncThread(object):
 
-    def __init__(self, target=None, args=(), kwargs={}, daemon=False):
+    def __init__(self, target=None, args=None, kwargs=None, daemon=False):
```

Comment:
@patrickBerry, `()` is already immutable, so that half of this PR is invalid.

The `kwargs={}` half might actually be valid, in general, since that dict instance is essentially a module global that anyone who can get a reference to it can mutate for everyone else.
```python
>>> class Foo:
...     def __init__(self, kwargs={}):
...         self.kwargs = kwargs
... 
>>> f1 = Foo()
>>> f2 = Foo()
>>> f2.kwargs['global'] = True
>>> f1.kwargs
{'global': True}
```
But rather than changing this to `kwargs=None`, it seems cleaner to use `**kwargs`. Then each call that doesn't supply any kwargs results in `kwargs` getting bound to a new empty dict, resolving this concern, and the code on line 73 doesn't actually have to change.